src/hooks/validation/ValidationService.ts

```diff
@@ -0,0 +1,19 @@
+import { injectable } from "tsyringe";
```

Comment:
It's only a question: Why do you want use DI system in cli? I like DI, but only when it's needed, I mean for huge app/source code. Of course, we know that cli will be in the future grow and grow, but at the moment using DI is a some kind of boilerplate :) We can discuss about it in the future, at the moment it can stay. I don't want to block you.

Also be aware: `tsyringe` or `typedi` and also another "simple" DI systems in TS has a one big problem (for me). They works with one global container, so usually for growing app it's a killer. We can choose another one after discussion. Could you create issue for that? :)

Reply:
This first "feature" is kind a boilerplate itself so the main idea of adding a DI is to make it easier for future and also because I consider that it is easier to test and mock the different layers. 
If there are better DI libraries I'm open to change tsyringe, but this one was choosen because is the one I know 😅

Reply:
> This first "feature" is kind a boilerplate itself so the main idea of adding a DI is to make it easier for future and also because I consider that it is easier to test and mock the different layers.

I understand, but be aware that at the beginning it will be boilerplate for some people :)

> If there are better DI libraries I'm open to change tsyringe, but this one was choosen because is the one I know 😅

I know about (mentioned) `typedi`, `tsyringe`, `awilix`, `typescript-ioc` but I wouldn't like to use them. They are very simple and some of them have also problem with simple injection like property injection etc. At the moment I only consider to use the `inversify` but it's also not good. In my spare time I also write my own DI system based on Angular approach with modules and maybe we will try to use my library.

Reply:
I will create an issue to take a look at it at the future 😛